config/defaults.mk

```diff
@@ -24,7 +24,7 @@ EGREP_BIN = $(shell command -v egrep 2> /dev/null)
 CXX = g++
 MPICXX = mpicxx
 
-BASE_FLAGS  = -std=c++11
+BASE_FLAGS  = -std=c++17
```

Comment:
This type of change will require a lengthy internal discussion. Is this something needed by this PR or can we revert it?

Reply:
I can remove this one. I just did it for convenience to use an optional instead of a maybe-nullptr.